### PR TITLE
Add file upload

### PR DESCRIPTION
**Pull request by rayshan**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,22 +1,138 @@
...
+            method: "POST",
+            headers: new Headers({'Content-Type': 'application/json'}),
+            body: JSON.stringify(fileMetadata),
+        };
+        return fetch(uploadUrlSigningApi, fetchOptions)
+            .then(response => response.json())
+            .then(json => json.uploadURL);
+    }
+
+    static uploadFileWithSignedUrl(fileBlob, signedUrl) {
```
Comment (pavitrabhalla) [inline src/App.js]: Are we signing using our root AWS keys?

File: src/App.js [hunk 2]
```diff
@@ -1,22 +1,138 @@
...
+            const file = this.state.file;
+            const reader = new FileReader();
+            reader.onload = () =>
+                App.signedUrlFrom(
+                    {
+                        filename: this.filename,
+                        type: file.type,
+                        metadata: {email: this.state.email},
+                    },
+                    "https://9t9r1qnsi1.execute-api.us-west-2.amazonaws.com/dev/requestUploadURL",
```
Comment (pavitrabhalla) [inline src/App.js]: Is this your dev lambda URL? Would we need to get a separate on to use in prod?

File: package.json [hunk 3]
```diff
@@ -5,12 +5,23 @@
...
+    "watch-css": "npm run build-css && node-sass-chokidar --include-path ./node_modules src/ -o src/ --watch --recursive",
+    "start-js": "BROWSER=none react-scripts start",
+    "start": "BROWSER=none npm-run-all -p watch-css start-js",
+    "build-js": "react-scripts build",
+    "build": "npm-run-all build-css build-js",
     "test": "react-scripts test --env=jsdom",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "predeploy": "npm run build",
+    "deploy": "netlify deploy"
```
[the remark this answers is not in the document]
Comment (rayshan) [reply package.json]: Oops, this shouldn't be there actually. It should be swapped with the heroku command. I'll remove it.

File: src/App.js (hunk 1 above)
Comment (rayshan) [reply src/App.js]: Yes, in lambda function. Keys are not embedded in the client.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (pavitrabhalla) [reply package.json]: I already have a static buildback heroku deploy command. We dont need anything added here for deploying/building

File: src/App.js (hunk 1 above)
Comment (pavitrabhalla) [reply src/App.js]: Very nice

**Pull request merged**